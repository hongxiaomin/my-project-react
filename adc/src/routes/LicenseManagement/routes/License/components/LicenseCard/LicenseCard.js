import React from 'react'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { Input, InputNumber, Icon, Radio } from 'antd'
import { connect } from 'react-redux'
import 'react-table/react-table.css'
import classes from './LicenseCard.scss'
import { trainingOrJudging, getLicense, activate, save } from '../../modules/'

const RadioGroup = Radio.Group
type Props = {
  dispatch: Function,
  item: Array,
}

export class LicenseCard extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isTrainingOrJudging: this.props.item.isTrainingDevice,
      defaultValue: '',
      inputString: '',
      name: this.props.item.name || '',
      isEdit: false,
    }
    this.tmp = []
  }

  componentWillReceiveProps(nextProps) {
    console.log('nextProps', this.state.isTrainingOrJudging, nextProps.item.isTrainingDevice)
    if (this.state.isTrainingOrJudging !== nextProps.item.isTrainingDevice) {
      this.setState({
        isTrainingOrJudging: nextProps.item.isTrainingDevice,
      })
    }
    this.setState({
      name: nextProps.item.name,
    })
  }

  onChange = (e) => {
    if (this.props.item.isActivate === false) {
      this.setState({
        isTrainingOrJudging: e.target.value,
      })
    } else {
      this.setState({
        isTrainingOrJudging: e.target.value,
      })
      this.props.dispatch(trainingOrJudging({ id: this.props.item.id,
        isTrainingDevice: e.target.value }))
    }
  }

  getLicense = () => {
    this.tmp = Object.assign([], this.props.item.deviceGpus)
    const tmp = {
      id: this.props.item.id,
      deviceGpus: this.tmp,
      deviceHash: this.props.item.deviceHash,
    }
    this.props.dispatch(getLicense(tmp))
  }

  handleChange = (type, keyIndex) => (e) => {
    this.tmp = Object.assign([], this.props.item.deviceGpus)
    this.tmp.forEach((item) => {
      if (keyIndex === item.id) {
        if (type === 'judging') {
          item.judgingServiceCount = e.toString()
          item.id = item.id.toString()
        } else {
          item.trainingServiceCount = e.toString()
          item.id = item.id.toString()
        }
      }
    })
  }
  blur = (type, keyIndex) => (e) => {
    this.tmp = Object.assign([], this.props.item.deviceGpus)
    this.tmp.forEach((item) => {
      if (keyIndex.toString() === item.id) {
        if (type === 'judging') {
          item.judgingServiceCount = e.target.value.toString()
        } else {
          item.trainingServiceCount = e.target.value.toString()
        }
        item.id = item.id.toString()
      }
    })
  }
  handleChangeOtherString = (e) => {
    this.setState({
      inputString: e.target.value,
    })
  }
  activate = () => {
    if (!this.props.item.license) return
    const tmp = {
      id: this.props.item.id.toString(),
      license: this.state.inputString,
      deviceGpus: this.tmp,
      deviceHash: this.props.item.deviceHash,
      isTrainingDevice: this.state.isTrainingOrJudging,
    }

    this.props.dispatch(activate(tmp))
  }

  edit = () => {
    this.setState({
      isEdit: !this.state.isEdit,
    })
  }

  save = () => {
    const tmp = {
      id: this.props.item.id.toString(),
      name: this.state.name,
    }
    this.props.dispatch(save(tmp, () => {
      this.setState({
        isEdit: false,
      })
    }))
  }

  closeSave = () => {
    this.setState({
      isEdit: false,
      name: this.props.item.name,
    })
  }
  handleChangeDeviceName = (e) => {
    this.setState({
      name: e.target.value,
    })
  }

  props: Props
  render() {
    console.log('training', this.state.isTrainingOrJudging)
    const { intl } = this.props

    return (
      <div style={{ boxShadow: '6px 6px #ccc', border: '1px solid #ccc', marginTop: '30px', borderRadius: '5px' }}>
        <div style={{ display: 'flex', backgroundColor: '#f7f7f7', justifyContent: 'space-between', paddingRight: '20px' }}>
          <div className={classes.top}><div><WrapperFormattedMessage id='device' />{this.props.item.id}</div>：
            {this.state.isEdit
              ? (<Input
                style={{
                  display: 'inline-block',
                  width: 'auto',
                  marginTop: '16px',
                }}
                value={this.state.name}
                onChange={this.handleChangeDeviceName}
              />)
              : <span title={this.state.name} style={{ width: '300px', overflow: 'hidden', whiteSpace: 'nowrap', textOverflow: 'ellipsis' }}>{this.state.name}</span>
            }
          </div>
          {this.state.isEdit
            ? <div><Icon
              type="check"
              style={{ width: '20px', height: '20px', margin: '20px 10px 0 0', fontSize: '20px', cursor: 'pointer' }}
              onClick={this.save}
              disabled={this.props.item.isActivate}
              title={intl.formatMessage({ id: 'Save' })}
            />
              <Icon
                type="close-circle-o"
                style={{ width: '20px', height: '20px', marginTop: '20px', fontSize: '20px', cursor: 'pointer' }}
                onClick={this.closeSave}
                disabled={this.props.item.isActivate}
                title={intl.formatMessage({ id: 'cancel' })}
              /></div>
            : <Icon
              type="edit"
              style={{ width: '20px', height: '20px', marginTop: '20px', fontSize: '20px', cursor: 'pointer' }}
              onClick={this.edit}
              disabled={this.props.item.isActivate}
              title={intl.formatMessage({ id: 'edit' })}
            />
          }
        </div>
        <div>
          <div className={classes.hash}>Device Hash：{this.props.item.deviceHash}</div>
          <div className={classes.num}>
            <div>
              {
                this.props.item.deviceGpus && this.props.item.deviceGpus.map((item, index) => {
                  return (
                    <div key={index} style={{ marginBottom: '10px', display: 'flex', justifyContent: 'start', alignItems: 'center' }}>
                      <div><WrapperFormattedMessage id='gpuId' />：{item.gpuId}</div>
                      <div style={{ marginLeft: '20px', display: 'flex', justifyContent: 'start', alignItems: 'center' }}>
                        <div><WrapperFormattedMessage id='judgingService' />：</div>
                        <InputNumber
                          min={0}
                          step={1}
                          disabled={this.props.item.isActivate}
                          defaultValue={item.judgingServiceCount}
                          onChange={this.handleChange('judging', item.id)}
                          onBlur={this.blur('judging', item.id)}
                          style={{ width: '100px', marginLeft: '20px' }}
                        />
                      </div>
                      <div style={{ marginLeft: '20px', display: 'flex', justifyContent: 'start', alignItems: 'center' }}>
                        <div><WrapperFormattedMessage id='trainingService' />：</div>
                        <InputNumber
                          min={0}
                          step={1}
                          disabled={this.props.item.isActivate}
                          defaultValue={item.trainingServiceCount}
                          onChange={this.handleChange('training', item.id)}
                          onBlur={this.blur('training', item.id)}
                          style={{ width: '100px', marginLeft: '20px' }}
                        />
                      </div>
                    </div>
                  )
                })
              }
            </div>
          </div>
          {
            this.props.item.isActivate === true ? <div /> :
            <div>
              <div className={classes.licen}>
                <div><WrapperFormattedMessage id='userAndLicenseManagement.licenseCard.serialNumber' />：{this.props.item.license}</div>
                <button
                  style={{ height: '30px' }}
                  className={'pt-button pt-intent-primary'}
                  onClick={this.getLicense}
                  disabled={this.props.item.isActivate}
                >
                  <WrapperFormattedMessage id='userAndLicenseManagement.licenseCard.getSerialNumber' />
                </button>
              </div>
              <div style={{ margin: '0px 20px 20px 20px', borderBottom: '1px solid #ccc', paddingBottom: '20px' }}>
                <span><WrapperFormattedMessage id='userAndLicenseManagement.licenseCard.getSerialNumber' />：</span>
                <Input
                  style={{
                    display: 'inline-block',
                    width: '60%',
                    marginTop: '16px',
                  }}
                  onChange={this.handleChangeOtherString}
                />
              </div>
            </div>
          }
          <div className={classes.mode}>
            <div><WrapperFormattedMessage id='userAndLicenseManagement.licenseCard.operationMode' />：</div>
            <div>
              <RadioGroup
                onChange={this.onChange}
                // disabled={this.props.item.isActivate}
                value={this.state.isTrainingOrJudging}
              >
                <Radio value={true}><WrapperFormattedMessage id='userAndLicenseManagement.licenseCard.training' /></Radio>
                <Radio value={false}><WrapperFormattedMessage id='userAndLicenseManagement.licenseCard.judging' /></Radio>
              </RadioGroup>
            </div>
          </div>
          <div className={classes.radio}>
            <button
              className={'pt-button pt-intent-primary'}
              onClick={this.activate}
              disabled={this.props.item.isActivate}
            >
              <WrapperFormattedMessage id='userAndLicenseManagement.licenseCard.activate' />
            </button>
          </div>
        </div>
      </div>
    )
  }
}
export default connect(
  ({ auth, intl: { locale } }) => ({
    locale,
    token: auth.token,
    // modelSettingList: state.licenseManagement,
  }),
)(injectIntl(LicenseCard))
