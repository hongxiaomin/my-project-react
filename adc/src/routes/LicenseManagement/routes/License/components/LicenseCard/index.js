import LicenseCard from './LicenseCard'

export default LicenseCard
