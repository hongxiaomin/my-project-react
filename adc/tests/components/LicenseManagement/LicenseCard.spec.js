import React from 'react'
import { shallow, mount } from 'enzyme'
import _ from 'lodash'
import { LicenseCard } from 'routes/LicenseManagement/routes/License/components/LicenseCard/LicenseCard'

describe('(Component) edit is true', () => {
  const dispatch = jest.fn()
  const item = [{ deviceGpu: [{ id: 1 }], name: 'a' }]
  it('render normally', () => {
    const nativeEvent = { nativeEvent: { stopImmediatePropagation: _.noop } }
    const e = { target: { value: 'name' } }
    const save = jest.fn()
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    wrapper.find('Icon').at(0).simulate('click', nativeEvent)
    expect(wrapper.state('isEdit')).toEqual(true)
    wrapper.find('Icon').at(1).simulate('click', e)
    const tmp = {}
    dispatch(save(tmp, () => {}))
    expect(wrapper.state('isEdit')).toEqual(false)
    // wrapper.find('InputNumber').simulate('change', )

    expect(wrapper).toMatchSnapshot()
  })
  it('render name state', () => {
    const e = { nativeEvent: { stopImmediatePropagation: _.noop }, target: { value: 'name' } }
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    wrapper.find('Icon').at(0).simulate('click', e)
    expect(wrapper.state('isEdit')).toEqual(true)
    wrapper.find('Input').at(0).simulate('change', e)
    expect(wrapper.state('name')).toEqual('name')
    expect(wrapper).toMatchSnapshot()
  })
})

describe('(Component) edit is false', () => {
  const dispatch = jest.fn()
  const item = { deviceGpu: [{ id: 1 }], name: 'a' }
  it('render button getLicense', () => {
    const nativeEvent = { nativeEvent: { stopImmediatePropagation: _.noop } }
    const e = { target: { value: 'name' } }
    const save = jest.fn()
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    wrapper.find('button').at(0).simulate('click')
    // expect(wrapper.state('name')).toEqual('name')
    expect(wrapper).toMatchSnapshot()
  })
  it('render input handleChangeOtherString', () => {
    // const nativeEvent = { nativeEvent: { stopImmediatePropagation: _.noop } }
    const e = { target: { value: 'name' } }
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    wrapper.find('Input').at(0).simulate('change', e)
    expect(wrapper.state('inputString')).toEqual('name')
    expect(wrapper).toMatchSnapshot()
  })
  it('render radioGroup', () => {
    // const nativeEvent = { nativeEvent: { stopImmediatePropagation: _.noop } }
    const e = { target: { value: 'name' } }
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    wrapper.find('RadioGroup').at(0).simulate('change', e)
    expect(wrapper.state('isTrainingOrJudging')).toEqual('name')
    expect(wrapper).toMatchSnapshot()
  })
  it('render activate', () => {
    const nativeEvent = { nativeEvent: { stopImmediatePropagation: _.noop } }
    const e = { target: { value: 'name' } }
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    wrapper.find('button').at(1).simulate('click', nativeEvent)
    // expect(wrapper.state('isTrainingOrJudging')).toEqual('name')
    expect(wrapper).toMatchSnapshot()
  })
})
describe('(Component) isactivate is true', () => {
  const dispatch = jest.fn()
  const item = { deviceGpus: [{ id: 1 }], name: 'a', isActivate: true }
  it('render', () => {
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render input number', () => {
    const e = 1
    const wrapper = shallow(
      <LicenseCard
        dispatch={dispatch}
        item={item}
      />,
    )
    wrapper.find('InputNumber').at(0).simulate('change', e)
    expect(wrapper.state('count')).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
