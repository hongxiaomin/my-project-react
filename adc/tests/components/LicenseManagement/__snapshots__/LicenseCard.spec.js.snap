// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) edit is false render activate 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <span
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "300px",
          }
        }
        title="a"
      >
        a
      </span>
    </div>
    <Icon
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "height": "20px",
          "marginTop": "20px",
          "width": "20px",
        }
      }
      title="Edit"
      type="edit"
    />
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div />
    </div>
    <div>
      <div
        className="licen"
      >
        <div>
          Serial Number：
        </div>
        <button
          className="pt-button pt-intent-primary"
          onClick={[Function]}
          style={
            Object {
              "height": "30px",
            }
          }
        >
          Get Serial Number
        </button>
      </div>
      <div
        style={
          Object {
            "borderBottom": "1px solid #ccc",
            "margin": "0px 20px 20px 20px",
            "paddingBottom": "20px",
          }
        }
      >
        <span>
          License：
        </span>
        <Input
          disabled={false}
          onChange={[Function]}
          prefixCls="ant-input"
          style={
            Object {
              "display": "inline-block",
              "marginTop": "16px",
              "width": "60%",
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) edit is false render button getLicense 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <span
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "300px",
          }
        }
        title="a"
      >
        a
      </span>
    </div>
    <Icon
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "height": "20px",
          "marginTop": "20px",
          "width": "20px",
        }
      }
      title="Edit"
      type="edit"
    />
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div />
    </div>
    <div>
      <div
        className="licen"
      >
        <div>
          Serial Number：
        </div>
        <button
          className="pt-button pt-intent-primary"
          onClick={[Function]}
          style={
            Object {
              "height": "30px",
            }
          }
        >
          Get Serial Number
        </button>
      </div>
      <div
        style={
          Object {
            "borderBottom": "1px solid #ccc",
            "margin": "0px 20px 20px 20px",
            "paddingBottom": "20px",
          }
        }
      >
        <span>
          License：
        </span>
        <Input
          disabled={false}
          onChange={[Function]}
          prefixCls="ant-input"
          style={
            Object {
              "display": "inline-block",
              "marginTop": "16px",
              "width": "60%",
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) edit is false render input handleChangeOtherString 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <span
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "300px",
          }
        }
        title="a"
      >
        a
      </span>
    </div>
    <Icon
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "height": "20px",
          "marginTop": "20px",
          "width": "20px",
        }
      }
      title="Edit"
      type="edit"
    />
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div />
    </div>
    <div>
      <div
        className="licen"
      >
        <div>
          Serial Number：
        </div>
        <button
          className="pt-button pt-intent-primary"
          onClick={[Function]}
          style={
            Object {
              "height": "30px",
            }
          }
        >
          Get Serial Number
        </button>
      </div>
      <div
        style={
          Object {
            "borderBottom": "1px solid #ccc",
            "margin": "0px 20px 20px 20px",
            "paddingBottom": "20px",
          }
        }
      >
        <span>
          License：
        </span>
        <Input
          disabled={false}
          onChange={[Function]}
          prefixCls="ant-input"
          style={
            Object {
              "display": "inline-block",
              "marginTop": "16px",
              "width": "60%",
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) edit is false render radioGroup 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <span
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "300px",
          }
        }
        title="a"
      >
        a
      </span>
    </div>
    <Icon
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "height": "20px",
          "marginTop": "20px",
          "width": "20px",
        }
      }
      title="Edit"
      type="edit"
    />
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div />
    </div>
    <div>
      <div
        className="licen"
      >
        <div>
          Serial Number：
        </div>
        <button
          className="pt-button pt-intent-primary"
          onClick={[Function]}
          style={
            Object {
              "height": "30px",
            }
          }
        >
          Get Serial Number
        </button>
      </div>
      <div
        style={
          Object {
            "borderBottom": "1px solid #ccc",
            "margin": "0px 20px 20px 20px",
            "paddingBottom": "20px",
          }
        }
      >
        <span>
          License：
        </span>
        <Input
          disabled={false}
          onChange={[Function]}
          prefixCls="ant-input"
          style={
            Object {
              "display": "inline-block",
              "marginTop": "16px",
              "width": "60%",
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          defaultValue="name"
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) edit is true render name state 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <Input
        disabled={false}
        onChange={[Function]}
        prefixCls="ant-input"
        style={
          Object {
            "display": "inline-block",
            "marginTop": "16px",
            "width": "auto",
          }
        }
        type="text"
        value="name"
      />
    </div>
    <div>
      <Icon
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
            "height": "20px",
            "margin": "20px 10px 0 0",
            "width": "20px",
          }
        }
        title="Save"
        type="check"
      />
      <Icon
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
            "height": "20px",
            "marginTop": "20px",
            "width": "20px",
          }
        }
        title="cancel"
        type="close-circle-o"
      />
    </div>
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div />
    </div>
    <div>
      <div
        className="licen"
      >
        <div>
          Serial Number：
        </div>
        <button
          className="pt-button pt-intent-primary"
          onClick={[Function]}
          style={
            Object {
              "height": "30px",
            }
          }
        >
          Get Serial Number
        </button>
      </div>
      <div
        style={
          Object {
            "borderBottom": "1px solid #ccc",
            "margin": "0px 20px 20px 20px",
            "paddingBottom": "20px",
          }
        }
      >
        <span>
          License：
        </span>
        <Input
          disabled={false}
          onChange={[Function]}
          prefixCls="ant-input"
          style={
            Object {
              "display": "inline-block",
              "marginTop": "16px",
              "width": "60%",
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) edit is true render normally 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <span
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "300px",
          }
        }
      />
    </div>
    <Icon
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "height": "20px",
          "marginTop": "20px",
          "width": "20px",
        }
      }
      title="Edit"
      type="edit"
    />
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div />
    </div>
    <div>
      <div
        className="licen"
      >
        <div>
          Serial Number：
        </div>
        <button
          className="pt-button pt-intent-primary"
          onClick={[Function]}
          style={
            Object {
              "height": "30px",
            }
          }
        >
          Get Serial Number
        </button>
      </div>
      <div
        style={
          Object {
            "borderBottom": "1px solid #ccc",
            "margin": "0px 20px 20px 20px",
            "paddingBottom": "20px",
          }
        }
      >
        <span>
          License：
        </span>
        <Input
          disabled={false}
          onChange={[Function]}
          prefixCls="ant-input"
          style={
            Object {
              "display": "inline-block",
              "marginTop": "16px",
              "width": "60%",
            }
          }
          type="text"
        />
      </div>
    </div>
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) isactivate is true render 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <span
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "300px",
          }
        }
        title="a"
      >
        a
      </span>
    </div>
    <Icon
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "height": "20px",
          "marginTop": "20px",
          "width": "20px",
        }
      }
      title="Edit"
      type="edit"
    />
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "start",
              "marginBottom": "10px",
            }
          }
        >
          <div>
            GPU ID：
          </div>
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "start",
                "marginLeft": "20px",
              }
            }
          >
            <div>
              Service：
            </div>
            <InputNumber
              disabled={true}
              min={0}
              onBlur={[Function]}
              onChange={[Function]}
              prefixCls="ant-input-number"
              step={1}
              style={
                Object {
                  "marginLeft": "20px",
                  "width": "100px",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div />
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        disabled={true}
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;

exports[`(Component) isactivate is true render input number 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "5px",
      "boxShadow": "6px 6px #ccc",
      "marginTop": "30px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "display": "flex",
        "justifyContent": "space-between",
        "paddingRight": "20px",
      }
    }
  >
    <div
      className="top"
    >
      <div>
        Device
      </div>
      ：
      <span
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "300px",
          }
        }
        title="a"
      >
        a
      </span>
    </div>
    <Icon
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "fontSize": "20px",
          "height": "20px",
          "marginTop": "20px",
          "width": "20px",
        }
      }
      title="Edit"
      type="edit"
    />
  </div>
  <div>
    <div
      className="hash"
    >
      Device Hash：
    </div>
    <div
      className="num"
    >
      <div>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "start",
              "marginBottom": "10px",
            }
          }
        >
          <div>
            GPU ID：
          </div>
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "start",
                "marginLeft": "20px",
              }
            }
          >
            <div>
              Service：
            </div>
            <InputNumber
              disabled={true}
              min={0}
              onBlur={[Function]}
              onChange={[Function]}
              prefixCls="ant-input-number"
              step={1}
              style={
                Object {
                  "marginLeft": "20px",
                  "width": "100px",
                }
              }
              value="1"
            />
          </div>
        </div>
      </div>
    </div>
    <div />
    <div
      className="mode"
    >
      <div>
        Operation Mode：
      </div>
      <div>
        <RadioGroup
          disabled={false}
          onChange={[Function]}
        >
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={true}
          >
            Training
          </Radio>
          <Radio
            prefixCls="ant-radio"
            type="radio"
            value={false}
          >
            Judging
          </Radio>
        </RadioGroup>
      </div>
    </div>
    <div
      className="radio"
    >
      <button
        className="pt-button pt-intent-primary"
        disabled={true}
        onClick={[Function]}
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;
